x/params/subspace/subspace.go

```diff
@@ -154,51 +149,61 @@ func (s Subspace) checkType(store sdk.KVStore, key []byte, param interface{}) {
 	}
 
 	if pty != ty {
-		panic("Type mismatch with registered table")
+		return errors.New("Type mismatch with registered table")
 	}
+	return nil
 }
 
 // Set stores the parameter. It returns error if stored parameter has different type from input.
 // It also set to the transient store to record change.
-func (s Subspace) Set(ctx sdk.Context, key []byte, param interface{}) {
+func (s Subspace) Set(ctx sdk.Context, key []byte, param interface{}) error {
 	store := s.kvStore(ctx)
 
-	s.checkType(store, key, param)
+	if err := s.checkType(store, key, param); err != nil {
+		return err
+	}
 
 	bz, err := s.cdc.MarshalJSON(param)
```

Comment:
In my view, this should really fail loudly if `param` is really unmarshalable:

```suggestion
	bz := s.cdc.MustMarshalJSON(param)
```

Reply:
Why?

Reply:
I would not agree to panic here since this is a package for storage. If there is an error on storage layer, return the error and let main caller decide if we should panic or not (eg cmd layer).

Reply:
Agreed